server/src/com/thoughtworks/go/server/materials/MaterialDatabaseUpdater.java

```diff
@@ -87,7 +81,7 @@ public void updateMaterial(final Material material) throws Exception {
                     LOGGER.debug(String.format("[Material Update] Material repository not found, creating with latest revision from %s", material));
                 }
 
-                synchronized (cacheKeyForMaterial) {
+                synchronized (materialMutex) {
```

Comment:
I wonder whether it has anything to do with the "concurrent" updates allowed for non-auto-update materials that I mentioned.